### PR TITLE
Bug 1365345 - Modal Bug Entry Prototype

### PR DESCRIPTION
**Pull request by SebastinSanty**

### CODE REVIEW

**Comment by dylanwh** [comment]
I mean, that's one option. Another option would be to figure out how to handle the back/forward cache.

File: extensions/BugModal/lib/WebService.pm [hunk 1]
```diff
@@ -87,19 +87,24 @@ sub products {
...
-    return { components => \@components }
+    my $product = Bugzilla::Product->check( { name => $params->{product_name}, cache => 1 } );
+    $product = Bugzilla->user->can_enter_product( $product, 1 );
+    my @components = map {
+        {
+            name        => $_->name,
+            description => $_->description,
+        }
+    } @{ $product->components };
+    my @versions = map { { name => $_->name } } grep( $_->is_active, @{ $product->versions } );
```
Comment (dylanwh) [inline extensions/BugModal/lib/WebService.pm]: so both grep and map have two forms: `grep code, list` and `grep { code } list`
It's better to just always use the block style.
so make this:
```perl
my @versions = map { { name => $_->name } } grep { $_->is_active } @{ $product->versions };

File: extensions/BugModal/web/new_bug.js [hunk 2]
```diff
@@ -1,48 +1,89 @@
...
-                $('#product').append($('<option>', { value: 'Select Product', text: 'Select Product' }));
-                // populate select menus
-                $.each(data.products, function(key, value) {
-                    $('#product').append($('<option>', { value: value.name, text: value.name }));
-                });
+                callback(data.products);
             },
-            function() {}
+            function() {
+                callback();
```
Comment (dylanwh) [inline extensions/BugModal/web/new_bug.js]: hmm, what does calling callback() with no arguments actually do?

Comment (SebastinSanty) [inline extensions/BugModal/web/new_bug.js]: No need of that here, as the data is already loaded (or else it goes out through the bugzilla_ajax' error function).

File: extensions/BugModal/template/en/default/bug_modal/field.html.tmpl [hunk 3]
```diff
@@ -28,6 +28,7 @@
   #                 edit html instead of replacing it.  forces edit_only (default: false);
   # default: (string) default value (eg. used as a placeholder in user fields)
   # help: (string) optional url that describes the field (requires a label to be defined and visible)
+  # required: (boolean) if the field is required.
```
Comment (dylanwh) [inline extensions/BugModal/template/en/default/bug_modal/field.html.tmpl]: comment should note this is only implemented for FIELD_TYPE_FREETEXT

**Pull request merged**